source/controllers/UserController.ts

```diff
@@ -51,6 +51,33 @@ login.read = (req, res) => {
 	}
 };
 
+const userpage = user.subcontroller('profile', ['id']);
+
+userpage.read = async (req, res) => {
+	const userpageId = req.params.id;
+	if (userpageId !== null && userpageId !== undefined) {
+		const usernameVar = await User.findByPk(userpageId);
+		if (usernameVar !== null && usernameVar !== undefined) {
+			const userlistings = await Listing.findAll({ where: { userId: usernameVar.id } });
+			if (res.locals.user !== null && res.locals.user !== undefined) {
+				const user: User = res.locals.user;
+				if (user.id === userpageId) {
+					if (res.locals.lang !== undefined) {
+						res.render('pages/user/userpage.ejs', { username: `${res.locals.lang.userpage.welcome},${usernameVar.username}`, constants: headerConstants, userstatus_page: `/user/profile/${user.id}`, userstatus_name: user.username, userlistings, currentuserid: user.id, authorid: usernameVar.id });
+					}
+				}
+				if (user.id !== userpageId) {
+					res.render('pages/user/userpage.ejs', { username: usernameVar.username, constants: headerConstants, userstatus_page: `/user/profile/${user.id}`, userstatus_name: user.username, userlistings, currentuserid: user.id, authorid: 'abc' });
+				}
+			} else {
+				res.render('pages/user/userpage.ejs', { username: usernameVar, constants: headerConstants, userstatus_page: '/user/login', userlistings, currentuserid: null, authorid: 'abc' });
+			}
+		}
+	} else {
+		res.redirect('/whoops');
```

Comment:
This page does not seem to exist, though?

Reply:
If meant to redirect to 404 this is a valid option bc the router will fallback to 404 but there should be specific route to 404 added

Reply:
changed to sendstatus(404) in latest commit